bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TableValueImpl.java

```diff
@@ -110,7 +110,7 @@ public TableValueImpl(TableType type) {
         this.indexToKeyMap = new LinkedHashMap<>();
         this.fieldNames = type.getFieldNames();
         this.keyValues = new LinkedHashMap<>();
-        if (type.getFieldNames() != null) {
+        if (type.getFieldNames() != null && type.getFieldNames().length != 0) {
```

Comment:
Wouldnt this be much simpler if we get rid of the null check and use an empty list for `fieldNames` and `type.fieldNameList` everywhere? then we don't need to check for both null and emptiness.

Reply:
Updated.